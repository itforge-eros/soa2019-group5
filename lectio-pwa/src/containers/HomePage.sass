.home--app_bar
	top: auto
	bottom: 0