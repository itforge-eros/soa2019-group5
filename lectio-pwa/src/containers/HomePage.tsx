import React, { Component, Fragment } from 'react';
import { Toolbar, Typography, AppBar, Fab, IconButton } from '@material-ui/core';
import AddIcon from '@material-ui/icons/Add';
import SettingsIcon from '@material-ui/icons/Settings';
import SearchIcon from '@material-ui/icons/Search';
import Header from '../components/Header';
import './HomePage.sass';

const styles = {
	appBar: {
		top: 'auto',
		bottom: '0'
	},
	toolbar: {
		alignItems: 'center',
		justifyContent: 'space-between'
	},
	fab: {
		position: 'absolute' as 'absolute',
		top: -30,
		left: 0,
		right: 0,
		margin: '0 auto'
	}
}

class HomePage extends Component {
  render() {
		return (
			<Fragment>
				<Header>
					<Typography variant="h5">Recordings</Typography>
				</Header>
				<AppBar position="fixed" color="primary" style={styles.appBar}>
					<Toolbar style={styles.toolbar}>
						<IconButton color="inherit" aria-label="Open Settings">
							<SettingsIcon />
						</IconButton>
						<Fab color="secondary" aria-label="Add" style={styles.fab}>
							<AddIcon />
						</Fab>
						<IconButton color="inherit" aria-label="Open Settings">
							<SearchIcon />
						</IconButton>
					</Toolbar>
				</AppBar>
			</Fragment>
		)
  }
}

export default HomePage;
