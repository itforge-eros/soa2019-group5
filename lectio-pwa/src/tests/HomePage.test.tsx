import React from 'react';
import renderer from 'react-test-renderer';
import HomePage from "../containers/HomePage";

it('renders HomePage with no internet connection', () => {
	const tree = renderer
		.create(
			<HomePage/>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
