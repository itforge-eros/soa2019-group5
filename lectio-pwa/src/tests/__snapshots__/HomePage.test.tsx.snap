// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HomePage with no internet connection 1`] = `
Array [
  <div
    className="header"
  >
    <h4
      className="MuiTypography-root-1 MuiTypography-h4-16"
      style={
        Object {
          "fontWeight": 600,
        }
      }
    >
      Recordings
    </h4>
  </div>,
  <div
    className="MuiLinearProgress-root-37 MuiLinearProgress-colorPrimary-38 MuiLinearProgress-indeterminate-41"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar-47 MuiLinearProgress-barColorPrimary-48 MuiLinearProgress-bar1Indeterminate-50"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar-47 MuiLinearProgress-barColorPrimary-48 MuiLinearProgress-bar2Indeterminate-53"
      style={Object {}}
    />
  </div>,
  <ul
    className="MuiList-root-55 MuiList-padding-56 list"
  />,
  <header
    className="MuiPaper-root-68 MuiPaper-elevation4-74 MuiAppBar-root-59 MuiAppBar-positionFixed-60 MuiAppBar-colorDefault-65 mui-fixed"
    style={
      Object {
        "bottom": "0",
        "top": "auto",
      }
    }
  >
    <div
      className="MuiToolbar-root-95 MuiToolbar-regular-97 MuiToolbar-gutters-96"
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "space-between",
        }
      }
    >
      <button
        aria-label="Open settings"
        className="MuiButtonBase-root-105 MuiIconButton-root-99 MuiIconButton-colorInherit-100"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-104"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-108"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-143"
        />
      </button>
      <button
        aria-label="Add a new memo"
        className="MuiButtonBase-root-105 MuiFab-root-117 MuiFab-primary-119"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": "#0062F5",
            "color": "#f4f4f4",
            "left": 0,
            "margin": "0 auto",
            "position": "absolute",
            "right": 0,
            "top": -30,
          }
        }
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiFab-label-118"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-108"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-143"
        />
      </button>
      <button
        aria-label="Search for memos"
        className="MuiButtonBase-root-105 MuiIconButton-root-99 MuiIconButton-colorInherit-100"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-104"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-108"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-143"
        />
      </button>
    </div>
  </header>,
]
`;
